crates/sparrow-compiler/src/types/inference.rs

```diff
@@ -543,7 +543,7 @@ fn least_upper_bound_data_type(a: DataType, b: &DataType) -> Option<DataType> {
 /// If the `concrete` type already satisfies this constraint, it is
```

Comment:
Maybe look at some of the comments here and/or search. Eg., "this type already satisfies this constraint" should be "this type is part of the type class", etc.

Reply:
Yeah, I'll probably add this to the next PR since I missed many in comments